algoliasearch/Transport.go

```diff
@@ -100,13 +96,14 @@ func (t *Transport) EncodeParams(params interface{}) string {
 	v := url.Values{}
 	if params != nil {
 		for key, value := range params.(map[string]interface{}) {
-			if reflect.TypeOf(value).Name() == "string" {
+			switch value.(type) {
```

Comment:
Here you can simply put `switch value := value.(type)`, so that you won't need to cast `value` in the case statements. More details here: https://golang.org/doc/effective_go.html#type_switch